docs/javascripts/downloads.js

```diff
@@ -0,0 +1,176 @@
+(function() {
+  const downloadsTableIdentifier = document.getElementById('downloads-table-identifier');
+
+  if (downloadsTableIdentifier) {
+    const appId = '52e3297d-ceeb-45c9-bbe4-6069fe920760';
+    const services = [
+      {
+        service: '<a href="../services/qix-engine/introduction/">Qlik Associative Engine</a>',
+        feature: 'The powerful associative indexing engine from Qlik and the foundation of Qlik Core.',
+        link: '<a href="https://hub.docker.com/r/qlikcore/engine">qlikcore/engine</a>',
+        apis: [
+          'EngineAPI',
+          'EngineRestAPI',
+          'ScriptLanguageAPI',
+        ],
+      }, {
+        service: '<a href="../services/licenses/">Licenses</a>',
+        feature: 'Service required to run Qlik Associative Engine with a paid license',
+        link: '<a href="https://hub.docker.com/r/qlikcore/licenses">qlikcore/licenses</a>',
+        apis: [
+          'LicenseServiceAPI',
+        ],
+      }, {
+        service: '<a href="../services/mira/">Mira</a>',
+        feature: 'Qlik Associative Engine discovery service.',
+        link: '<a href="https://hub.docker.com/r/qlikcore/mira">qlikcore/mira</a><br>(source code: <a href="https://github.com/qlik-oss/mira">qlik-oss/mira</a>)',
+        apis: [
+          'MiraAPI',
+        ],
+      },
+    ];
+
+    const loader = document.createElement('div');
+    loader.className = 'dots';
+    for (let i = 0; i < 7; i++) {
+      const dot = document.createElement('div');
+      dot.className = 'dots__dot';
+      loader.appendChild(dot);
+    }
+
+    const table = document.getElementsByTagName('table')[0];
+    const lastBodyColumn = table.querySelectorAll('tr td:last-child');
+    lastBodyColumn.forEach(cell => cell.appendChild(loader.cloneNode(true)));
+
+    const schema = {
+      "structs": {
+        "Global": {
+          "OpenDoc": {
+            "In": [{ "Name": "qDocName", "DefaultValue": "" }, { "Name": "qUserName", "DefaultValue": "", "Optional": true }, { "Name": "qPassword", "DefaultValue": "", "Optional": true }, { "Name": "qSerial", "DefaultValue": "", "Optional": true }, { "Name": "qNoData", "DefaultValue": false, "Optional": true }],
+            "Out": []
+          }
+        },
+        "Doc": {
+          "CreateSessionObject": {
+            "In": [{ "Name": "qProp", "DefaultValue": { "qInfo": { "qId": "", "qType": "" }, "qExtendsId": "", "qMetaDef": {} } }],
+            "Out": []
+          },
+        },
+        "GenericObject": {
+          "GetLayout": {
+            "In": [],
+            "Out": [{ "Name": "qLayout" }]
+          }
+        }
+      },
+    };
+
+    enigma.create({
+      schema,
+      url: `wss://branch.qlik.com/anon/app/${appId}`,
+    }).open()
+    .then(global => global.openDoc(appId))
+    .then(doc => {
+      doc.createSessionObject({
```

Comment:
You are not returning the promise chain here, so you're now creating two promise chains (and missing error handling [no `.catch()`])